source/extensionPoints.py

```diff
@@ -119,16 +124,48 @@ def myFunc(a=None, b=None):
 		callWithSupportedKwargs(myFunc, a=1, b=2, c=3)
 	Instead of raising a TypeError, myFunc will simply be called like this:
 		myFunc(a=1, b=2)
+
+	The `func` function arguments do not need to have default values, and can take kwargs to capture all arguments.
+	See tests/unit/test_extensionPoints.py:TestCallWithSupportedKwargs for examples.
+
+	An exception is raised if:
+	- the number of positional arguments given can not be received by `func`.
+	- parameters required (parameters declared with no default value) by `func` are not supplied.
 	"""
 	spec = inspect.getargspec(func)
+
+	# Ensure that we can provide all arguments without defaults expected by the handler are provided.
+	# `defaults` is a tuple of default argument values or None if there are no default arguments;
+	# if this tuple has n elements, they correspond to the last n elements listed in args.
+	numExpectedArgsWithDefaults = len(spec.defaults) if spec.defaults else 0
+
+	# some handlers are member functions, discard "self" because it is passed implicitly.

```

Comment:
It might be better to use the term "instance method" rather than "member function" here, just because I see that more in the Python world. Also, this won't work for class methods, where the first argument is normally named "cls". I don't know if that's a scenario we need to care about, though.